pyrk/th_component.py

```diff
@@ -83,33 +131,232 @@ def update_temp(self, timestep, temp):
         :param timestep: the timestep at which to query the temperature
         :type timestep: int
         :param temp: the new tempterature
-        :type float: float, units of kelvin
+        :type float: float, dimensionless
         """
         self.T[timestep] = temp
         self.prev_t_idx = timestep
         return self.T[timestep]
 
-    def dtemp(self):
-        if self.prev_t_idx == 0:
-            return 0.0*units.kelvin
-        else:
-            return (self.T[self.prev_t_idx] - self.T[self.prev_t_idx-1])
+    def dtemp(self, timestep, T0_timestep):
+        T0 = self.T[T0_timestep]
+        return self.T[timestep-1]-T0
 
-    def temp_reactivity(self):
-        return self.alpha_temp*self.dtemp()
+    def temp_reactivity(self, timestep, T0_timestep):
+        '''calculate reactivity of a component from temperature feedback
+        :param timestep: the timestep at which to calculate reactivity feedback
+        :type timestep: int
+        :param T0_timestep: the timestep at which the temperature is used as
+        reference temperature
+        :type T0_timestep: int
+        '''
+        assert timestep > T0_timestep, """reference timestep T0_timestep %f should
+        be prior to the timestep %f for temp feedback calculation""" % (
+            T0_timestep, timestep)
+        return self.alpha_temp*self.dtemp(timestep, T0_timestep)
```

Comment:
As noted before, you shouldn't have to pass this T0_timestep information through this function, as T0_timestep should be known by the timer, which dtemp can access internally to the function call.